sdk/core/core/src/az_span_builder.c

```diff
@@ -61,3 +61,15 @@ AZ_NODISCARD az_result az_span_builder_replace(
   self->length = size_after_replace;
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result
+az_span_builder_append_zeros(az_span_builder * const self, size_t const size) {
+  AZ_CONTRACT_ARG_NOT_NULL(self);
+
+  az_mut_span const span = az_mut_span_take(az_mut_span_drop(self->buffer, self->length), size);
+  if (span.size != size) {
+    return AZ_ERROR_BUFFER_OVERFLOW;
+  }
+  az_mut_span_fill(span, 0);
+  return AZ_OK;
```

Comment:
don't we need to update builder length after appending all zeros?  otherwize, using `append...` as function name is misleading, because it setting buffer's content but we can still write on it right?

For this case, I would prefer:
1. create span from array
2. fill span with 0
3. create span builder

Reply:
It's a bug.